be/src/exprs/table_function/table_function.h

```diff
@@ -79,18 +79,18 @@ class TableFunction {
     virtual ~TableFunction() = default;
 
     //Initialize TableFunctionState
-    virtual Status init(const TFunction& fn, TableFunctionState** state) const = 0;
+    [[nodiscard]] virtual Status init(const TFunction& fn, TableFunctionState** state) const = 0;
 
     //Some preparations are made in prepare, such as establishing a connection or initializing initial values
-    virtual Status prepare(TableFunctionState* state) const = 0;
+    [[nodiscard]] virtual Status prepare(TableFunctionState* state) const = 0;
 
-    virtual Status open(RuntimeState* runtime_state, TableFunctionState* state) const = 0;
+    [[nodiscard]] virtual Status open(RuntimeState* runtime_state, TableFunctionState* state) const = 0;
 
     //Table function processing logic
     virtual std::pair<Columns, UInt32Column::Ptr> process(TableFunctionState* state) const = 0;
 
     //Release the resources constructed in init and prepare
-    virtual Status close(RuntimeState* runtime_state, TableFunctionState* context) const = 0;
+    [[nodiscard]] virtual Status close(RuntimeState* runtime_state, TableFunctionState* context) const = 0;
```

Comment:
can we change this return type to void? since all override implementation always return OK.

Reply:
It's better to keep it. For now, `JavaUDTFFunction` will use it:
```
Status JavaUDTFFunction::close(RuntimeState* runtime_state, TableFunctionState* state) const {
    auto promise = call_function_in_pthread(runtime_state, [state]() {
        delete state;
        return Status::OK();
    });
    RETURN_IF_ERROR(promise->get_future().get());
    return Status::OK();
}
```

And it's better to keep it for the extension?